commit 2396089719976577b93c8f6959ee389afae36280
Author: Amir Rajan <ar@amirrajan.net>
Date:   Tue Jul 15 20:03:45 2025 -0500

    [mRuby] Applied patch https://github.com/mruby/mruby/commit/2d8f4fa92

    Fixes the following issue:

    module Bar
      def test
        puts "Bar#test"
        super
      end
    end

    class Foo
      prepend Bar

      def test
        puts "Foo#test"
        1.times { puts "In block" }
        puts "Foo#test"
      end
    end

    Foo.new.test

diff --git a/mruby/src/proc.c b/mruby/src/proc.c
index 870d5ea16..e374babde 100644
--- a/mruby/src/proc.c
+++ b/mruby/src/proc.c
@@ -66,7 +66,8 @@ mrb_env_new(mrb_state *mrb, struct mrb_context *c, mrb_callinfo *ci, int nstacks
   struct REnv *e;
   mrb_int bidx;

-  e = (struct REnv*)mrb_obj_alloc(mrb, MRB_TT_ENV, tc);
+  e = (struct REnv*)mrb_obj_alloc(mrb, MRB_TT_ENV, NULL);
+  e->c = tc;
   MRB_ENV_SET_LEN(e, nstacks);
   bidx = ci->argc;
   if (bidx < 0) bidx = 2;
@@ -90,7 +91,7 @@ closure_setup(mrb_state *mrb, struct RProc *p)
     /* do nothing, because e is assigned already */
   }
   else if (up) {
-    struct RClass *tc = MRB_PROC_TARGET_CLASS(p);
+    struct RClass *tc = ci->u.target_class;

     e = mrb_env_new(mrb, mrb->c, ci, up->body.irep->nlocals, ci->stack, tc);
     ci->u.env = e;
